src/analysis/classic/hunter/marksman/modules/features/Abilities.ts

```diff
@@ -0,0 +1,195 @@
+import SPELLS from 'common/SPELLS/classic/hunter';
+import CoreAbilities from 'parser/core/modules/Abilities';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+import RACIALS from 'common/SPELLS/classic/racials';
+
+class Abilities extends CoreAbilities {
+  spellbook() {
+    return [
+      // Rotational
+      // KILL_SHOT is handled in shared KillShot.ts because it only is valid in execute phase
+      {
+        spell: SPELLS.AUTO_SHOT.id,
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: null,
+      },
+      {
+        spell: [SPELLS.SERPENT_STING.id, ...SPELLS.SERPENT_STING.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+      },
+      {
+        spell: [SPELLS.CHIMERA_SHOT.id],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+      },
+      {
+        spell: [SPELLS.EXPLOSIVE_TRAP.id, ...SPELLS.EXPLOSIVE_TRAP.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        // gcd tracking with delayed trigger of trap seems to be a problem
+        gcd: null,
+        // Each talent point(3) in resources reduces cooldown by 2 seconds
+        cooldown: 24,
+      },
+      {
+        spell: [SPELLS.ARCANE_SHOT.id, ...SPELLS.ARCANE_SHOT.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+      },
+      {
+        spell: [SPELLS.AIMED_SHOT.id, ...SPELLS.AIMED_SHOT.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+        cooldown: 10,
+      },
+      {
+        spell: [SPELLS.MULTI_SHOT.id, ...SPELLS.MULTI_SHOT.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+        cooldown: 10,
+      },
+      {
+        spell: [SPELLS.STEADY_SHOT.id, ...SPELLS.STEADY_SHOT.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+      },
+      {
+        spell: [SPELLS.IMPROVED_STEADY_SHOT.id],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: null,
+      },
```

Comment:
buffs like ISS don't go in abilities. only abilities that you cast go here.

Reply:
How do you get IMPROVED_STEADY_SHOT on the timeline then ? I have it in Buffs but its not on the UI if I remove it from Abilities

Reply:
Never mind. I was looking at a log that didn't have IMPROVED_STEADY_SHOT and got confused.